### PR TITLE
[CP-stable]Refactor route focus node creation

### PR DESCRIPTION
**Pull request by flutteractionsbot**
### Issue Link:
https://github.com/flutter/flutter/issues/148867

Cherry-picking https://github.com/flutter/flutter/pull/147390

### Changelog Description:
Fixes a focus issue that causes TextFields to not function after cupertino back swipes

### Impact Description:
iOS, macos, iOS web, macos web app that uses TextField

### Workaround:
disable cupertino back swipe

### Risk:
What is the risk level of this cherry-pick?

  - [x] Low
  - [ ] Medium
  - [ ] High

### Test Coverage:
Are you confident that your fix is well-tested by automated tests?

  - [x] Yes
  - [ ] No

### Validation Steps:
What are the steps to validate that this fix works?

TextField should not be broken after a cupertino back swipe

### CODE REVIEW

**Comment by jmagman** [comment]
@chunhtai or @gspencergoog can you fill out this cherry-pick PR description?  There didn't seem to be any follow-up to https://github.com/flutter/flutter/issues/149178#issuecomment-2138325911.

For future reference, the cherry-pick automated process with labels is at:
https://github.com/flutter/flutter/blob/master/docs/releases/Flutter-Cherrypick-Process.md#automatically-creates-a-cherry-pick-request

**Review by chunhtai — COMMENTED** [review]
LGTM if also LGT @gspencergoog

**Review by chunhtai — APPROVED** [review]

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by jmagman** [comment]
@itsjustkevin heads up this was merged instead of going through the release team process.

**Comment by techouse** [comment]
@jmagman @itsjustkevin should https://github.com/flutter/flutter/issues/149178 not get closed?

**Comment by jmagman** [comment]
> @jmagman @itsjustkevin should #149178 now get closed?

https://github.com/flutter/flutter/blob/master/docs/releases/Flutter-Cherrypick-Process.md#lifecycle-of-a-cherry-pick
> Once the cherry-pick has been added to a release, the release engineering team will close the cherry-pick issue.

**Comment by BazinC** [comment]
Hi flutter team, what's required next? how long does it usually take after the merge of CP-stable to release a stable hotfix version of Flutter? (3.22.2?)
Thank you :)
(This bug is so painful... Our team upgraded from 3.19 -> 3.22, then had to downgrade because of this in order to keep releasing our app, and having to keep developing for both version is not productive at all even with the help of FVM)

**Comment by christopherfujino** [comment]
> Hi flutter team, what's required next? how long does it usually take after the merge of CP-stable to release a stable hotfix version of Flutter? (3.22.2?) Thank you 🙂 (This regression is so painful... Our team upgraded from 3.19 -> 3.22, then had to downgrade because of this in order to keep releasing our app, and having to keep developing for both version is not productive at all even with the help of FVM)

Stable 3.22.2 is now out

File: packages/flutter/lib/src/widgets/routes.dart
Comment (quyenlv-unicloud) [inline packages/flutter/lib/src/widgets/routes.dart:1068]: Hi, do we need to dispose `ValueNotifier<bool>(false),`?

**Comment by itsjustkevin** [comment]
Hi @quyenlv-unicloud, if you are running into problems, please create a new issue referencing this PR over commenting.

**Pull request merged**
